### PR TITLE
Ethdb

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: backend/coins/eth/account.go [hunk 1]
```diff
@@ -147,6 +163,38 @@ func (account *Account) poll() {
...
+	if err != nil {
+		return nil, err
+	}
+	defer dbTx.Rollback()
+	pendingOutgoingTransactions, err := dbTx.PendingOutgoingTransactions()
+	if err != nil {
+		return nil, err
+	}
+
+	confirmedTxHashes := map[string]struct{}{}
```
Comment (KasparEtter) [inline backend/coins/eth/account.go]: Is `struct{}` preferable to `boolean`?

Comment (benma) [reply backend/coins/eth/account.go]: yes. boolean contains useless information (true or false doesn't mean anything here).

File: backend/coins/eth/transaction.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/coin"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// wrappedTransaction wraps an outgoing pending transaction and implements coin.Transaction.
+type wrappedTransaction struct {
+	tx *types.Transaction
+}
+
+// NumConfirmations implements coin.Transaction.
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/coins/eth/transaction.go]: Thx. Wonder why the linter didn't catch this :thinking:

**Comment by benma** [comment]
>  Why do you persist go-ethereum/core/types transactions instead of our own coin.Transaction?

coin.Transaction is just an interface. The concrete implementation of it in this case is exactly `types.Transaction`. I want to store the proper eth transaction so we can rebroadcast/bump the fee of it in the future.

> Why did you introduce an interface and a struct for the ETH database? (Not that I mind.)

Out of habit :smile: so another DB backend can be plugged in, or a mock can be used in tests.

Maybe move backend/db to backend/coins/btc/db in a future PR?

:+1:

File: backend/coins/eth/transaction.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (benma) [reply backend/coins/eth/transaction.go]: Fixed. Linter didn't care because the type itself is not exported.

**Comment by benma** [comment]
> I know you don't like TODOs in the code, I would probably have added some in backend/coins/eth/transaction.go where the implementation is still fake/mocked.

It's not faked or mocked, it's complete. Note it models an unconfirmed outgoing tx:

```Go
// wrappedTransaction wraps an outgoing pending transaction and implements coin.Transaction.
type wrappedTransaction struct {
```

**Comment by KasparEtter** [comment]
Ah, I see. A more descriptive name would have helped… 🙂

**Pull request merged**